backend/api/auth.py

```
@@ -0,0 +1,455 @@
+        else:
+            expire = datetime.utcnow() + timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+        
+        to_encode.update({"exp": expire, "type": "access"})
+        
+        try:
+            encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+            logger.info(f"Created access token for user: {data.get('sub')}")
+            return encoded_jwt
+        except Exception as e:
+            logger.error(f"Failed to create access token: {e}")
+            raise HTTPException(status_code=500, detail="Failed to create access token")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)